src/Lykke.Snow.FirebaseIntegration/Services/FcmIntegrationService.cs

```
@@ -0,0 +1,70 @@
+        private readonly string _credentialsFilePath;
+
+        public FcmIntegrationService(ILogger<FcmIntegrationService> logger, string credentialsFilePath)
+        {
+            _logger = logger;
+            _credentialsFilePath = credentialsFilePath ?? throw new ArgumentNullException(nameof(credentialsFilePath));
+
+            if(!System.IO.File.Exists(_credentialsFilePath))
+                throw new FirebaseCredentialsFileNotFoundException(attemptedPath: _credentialsFilePath);
+
+            Initialize();
+        }
+
+        public async Task<Result<string, MessagingErrorCode>> SendNotification(Message fcmMessage, string deviceToken)
+        {
+            try 
+            {
+                var response = await FirebaseMessaging.DefaultInstance.SendAsync(fcmMessage);
+
+                return new Result<string, MessagingErrorCode>(value: response);
+            }
+            catch(ArgumentNullException e)
+            {
+                throw new NotificationMessageIsNotValidException(fcmMessage: fcmMessage, innerException: e, additionalInfo: "Message argument was null");
+            }
+            catch(ArgumentException e)
+            {
+                throw new NotificationMessageIsNotValidException(fcmMessage: fcmMessage, innerException: e);
+            }
+            catch(FirebaseMessagingException e)
+            {
+                //TODO: Handle different kinds of e.MessagingErrorCodes
+                throw new CannotSendNotificationException(fcmMessage: fcmMessage, fcmErrorCode: e.MessagingErrorCode, innerException: e);
+            }
+        }
+
+        private void Initialize()
+        {
+            if(FirebaseMessaging.DefaultInstance != null)
+                return;
+
+            try
+            {
+                FirebaseApp.Create(new AppOptions() 
+                {
+                    Credential = GoogleCredential.FromFile(_credentialsFilePath)
+                });
+            }
+            catch(ArgumentException)
```

Why is it empty? So you actually just swallow the exception without any reaction.

---

It's because of the thing we discussed here https://github.com/LykkeBusiness/Lykke.Snow.Notifications/pull/3#discussion_r1138566577

FCM SDK throws `ArgumentException` if FirebaseApp already exists according to the documentation. So, why would we crash the app then? Shouldn't we just ignore that exception?

I here catch the `ArgumentException` that's being thrown upon already existing FirebaseApp and silently ignore it. What action do you think we should take here?

---

I see, okay. Maybe then better to add a comment what this empty catch is intended to cover.

---

Right, added some brief explanation here.

https://github.com/LykkeBusiness/Lykke.Snow.Notifications/pull/3/files#diff-03debefeb89350b851d2a5b8fd1e0c06e1d3cd6dc8a49653ece8e1994154a4f7R67